src/tdastro/sources/periodic_source.py

```diff
@@ -0,0 +1,62 @@
+from abc import ABC, abstractmethod
+
+from tdastro.base_models import PhysicalModel
+
+
+class PeriodicSource(PhysicalModel, ABC):
+    """A periodic source.
+
+    Attributes
+    ----------
+    period : `float`
+        The period of the source, in days.
+    epoch : `float`
```

Comment:
I agree, I'll keep "epoch" for the documentation.

"Epoch" is a common term in variable star / old fashioned transient astronomy, search for "epoch of maximum light" or "epoch of primary minimum".

Reply:
(This might be my bias on the wording.) "Epoch" sounds a more general term. It makes more sense if you add "of ~" after it to refer to the specific term.